src/diffusers/models/unet_rl.py

```diff
@@ -0,0 +1,135 @@
+# model adapted from diffuser https://github.com/jannerm/diffuser/blob/main/diffuser/models/temporal.py
+from dataclasses import dataclass
+from typing import Tuple, Union
+
+import torch
+import torch.nn as nn
+
+from diffusers.models.resnet import Downsample1D, ResidualTemporalBlock1D
+from diffusers.models.unet_1d_blocks import get_down_block
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..modeling_utils import ModelMixin
+from ..utils import BaseOutput
+from .embeddings import TimestepEmbedding, Timesteps
+
+
+@dataclass
+class ValueFunctionOutput(BaseOutput):
+    """
+    Args:
+        sample (`torch.FloatTensor` of shape `(batch, horizon, 1)`):
+            Hidden states output. Output of last layer of model.
+    """
+
+    sample: torch.FloatTensor
+
+
+class ValueFunction(ModelMixin, ConfigMixin):
+    @register_to_config
+    def __init__(
+        self,
+        in_channels=14,
+        down_block_types: Tuple[str] = (
+            "DownResnetBlock1D",
+            "DownResnetBlock1D",
+            "DownResnetBlock1D",
+            "DownResnetBlock1D",
+        ),
+        block_out_channels: Tuple[int] = (32, 64, 128, 256),
+        act_fn: str = "mish",
+        norm_num_groups: int = 8,
+        layers_per_block: int = 1,
+    ):
+        super().__init__()
+        time_embed_dim = block_out_channels[0] * 4
+        self.time_proj = Timesteps(num_channels=block_out_channels[0], flip_sin_to_cos=False, downscale_freq_shift=1)
+        self.time_mlp = TimestepEmbedding(
+            channel=block_out_channels[0], time_embed_dim=time_embed_dim, act_fn="mish", out_dim=block_out_channels[0]
+        )
+
+        self.blocks = nn.ModuleList([])
+        mid_dim = block_out_channels[-1]
+
+        output_channel = in_channels
+        for i, down_block_type in enumerate(down_block_types):
+            input_channel = output_channel
+            output_channel = block_out_channels[i]
+
+            down_block_type = down_block_types[i]
+            down_block = get_down_block(
+                down_block_type,
+                num_layers=layers_per_block,
+                in_channels=input_channel,
+                out_channels=output_channel,
+                temb_channels=block_out_channels[0],
+                add_downsample=True,
+            )
+            self.blocks.append(down_block)
+
+        ##
+        self.mid_block1 = ResidualTemporalBlock1D(mid_dim, mid_dim // 2, embed_dim=block_out_channels[0])
+        self.mid_down1 = Downsample1D(mid_dim // 2, use_conv=True)
+        ##
+        self.mid_block2 = ResidualTemporalBlock1D(mid_dim // 2, mid_dim // 4, embed_dim=block_out_channels[0])
+        self.mid_down2 = Downsample1D(mid_dim // 4, use_conv=True)
+        ##
+        fc_dim = mid_dim // 4
+        self.final_block = nn.ModuleList(
```

Comment:
generally remove modulelists (harder to debug)

Reply:
I'm okay with changing the `final_block` to something cleaner - whatever y'all decide